sway/commands/resize.c

```diff
@@ -0,0 +1,297 @@
+		sibling = parent->parent->children->items[i];
+
+		int sibling_pos = parallel_coord(sibling, axis);
+		int focused_pos = parallel_coord(focused, axis);
+		int parent_pos = parallel_coord(parent, axis);
+
+		if (sibling_pos != focused_pos) {
+			if (sibling_pos < parent_pos) {
+				double pixels = -1 * amount;
+				pixels /= minor_weight;
+				if (major_weight) {
+					container_recursive_resize(sibling, pixels / 2, major_edge);
+				} else {
+					container_recursive_resize(sibling, pixels, major_edge);
+				}
+			} else if (sibling_pos > parent_pos) {
+				double pixels = -1 * amount;
+				pixels /= major_weight;
+				if (minor_weight) {
+					container_recursive_resize(sibling, pixels / 2, minor_edge);
+				} else {
+					container_recursive_resize(sibling, pixels, minor_edge);
+				}
+			}
+		} else {
+			if (major_weight != 0 && minor_weight != 0) {
+				double pixels = amount;
+				pixels /= 2;
+				container_recursive_resize(parent, pixels, minor_edge);
+				container_recursive_resize(parent, pixels, major_edge);
+			} else if (major_weight) {
+				container_recursive_resize(parent, amount, major_edge);
+			} else if (minor_weight) {
+				container_recursive_resize(parent, amount, minor_edge);
+			}
+		}
+	}
+
+	arrange_windows(parent->parent, -1, -1);
+}
+
+static void resize(int amount, enum resize_axis axis, enum resize_unit unit) {
+	struct sway_container *current = config->handler_context.current_container;
+	if (unit == RESIZE_UNIT_DEFAULT) {
+		// Default for tiling; TODO floating should be px
+		unit = RESIZE_UNIT_PPT;
+	}
+
+	if (unit == RESIZE_UNIT_PPT) {
+		float pct = amount / 100.0f;
+		switch (axis) {
+		case RESIZE_AXIS_HORIZONTAL:
+			amount = (float)current->width * pct;
+			break;
+		case RESIZE_AXIS_VERTICAL:
+			amount = (float)current->height * pct;
+			break;
+		default:
+			sway_assert(0, "invalid resize axis");
+			return;
+		}
+	}
+
+	return resize_tiled(amount, axis);
+}
+
+struct cmd_results *cmd_resize(int argc, char **argv) {
+	struct sway_container *current = config->handler_context.current_container;
+	if (!current) {
+		return cmd_results_new(CMD_INVALID, "resize", "Cannot resize nothing");
+	}
+	if (current->type != C_VIEW && current->type != C_CONTAINER) {
+		return cmd_results_new(CMD_INVALID, "resize",
+				"Can only resize views/containers");
+	}
+	if (strcasecmp(argv[0], "set") == 0) {
+		// TODO
+		//return cmd_resize_set(argc - 1, &argv[1]);
+		return cmd_results_new(CMD_INVALID, "resize", "resize set unimplemented");
+	}
+	struct cmd_results *error;
+	if ((error = checkarg(argc, "resize", EXPECTED_AT_LEAST, 2))) {
+		return error;
+	}
+
+	// TODO: resize grow|shrink left|right|up|down
+
+	const char *usage = "Expected 'resize <shrink|grow> "
+		"<width|height> [<amount>] [px|ppt]'";
+
+	int multiplier = 0;
+	if (strcasecmp(*argv, "grow") == 0) {
+		multiplier = 1;
+		--argc; ++argv;
+	} else if (strcasecmp(*argv, "shrink") == 0) {
+		multiplier = -1;
+		--argc; ++argv;
```

Comment:
What about putting this after the end of the `else`

Reply:
aight